packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1687,8 +1688,31 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
                 summaryLogger,
             });
 
+            const summary = summarizeResult.summary.tree;
+            const summaryTree = summary[channelsTreeName] as ISummaryTree;
+            let all = 0;
+            let unreferenced = 0;
+            if (summaryTree.tree !== undefined) {
+                all = Object.keys(summaryTree.tree).length;
+                for(const key of Object.keys(summaryTree.tree)) {
+                    const dataStore = summaryTree.tree[key] as ISummaryTree;
+                    if (dataStore.unreferenced === true) {
+                        unreferenced ++;
+                    }
```

Comment:
I don't think this is a great way to count them. The cast: `as ISummaryTree` should be a red flag here.

I think I would prefer one of the following instead:
1. Count at the lower layers (data store context) and return that up.
2. Loop like this, but check each node first to make sure it is a tree, and then make sure that it contains a ".component" blob in it, only count those. Something like:
```ts
const maybeDataStore = summaryTree.tree[key];
if (maybeDataStore.type === SummaryType.Tree && maybeDataStore.tree[".component"] !== undefined) {
    all++;
    if (maybeDataStore.unreferenced === true) {
        unreferenced++;
    }
}
```

Reply:
We need to be clear on what we are trying to log here. @arinwt With your suggestion, aren't we only counting number of data stores whose summary is a tree and ignore handles? If so, we should report that accordingly.

@chensixx  Can we clearly articulate what additional information we are trying to log. Is it total and unreferenced data store count in this document or total and unreferenced data store count for data stores whose summary is a tree?
Another thing is the issue says we need to log number of DDSs but I don't see that in the code. Do we want similar telemetry for DDSs too?
We can get all of this information but depending on what the requirement, we may need to do more work to get them.

Reply:
I think we should log totalDataStoreCount, totalSummarizedDataStoreCount(data stores which are not handles), and then unreferencedSummarizedDataStoreCount. That way we will have all the info.

Reply:
@arinwt I think I'm only counting at the highest layer, hence `totalDataStoreCount = Object.keys(summaryTree.tree).length;` I have fixed the casting

Reply:
> We need to be clear on what we are trying to log here. @arinwt With your suggestion, aren't we only counting number of data stores whose summary is a tree and ignore handles? If so, we should report that accordingly.
> 
> @chensixx Can we clearly articulate what additional information we are trying to log. Is it total and unreferenced data store count in this document or total and unreferenced data store count for data stores whose summary is a tree?
> Another thing is the issue says we need to log number of DDSs but I don't see that in the code. Do we want similar telemetry for DDSs too?
> We can get all of this information but depending on what the requirement, we may need to do more work to get them.

This is a good point; we should really take option (1) and do the counting in the datastore contexts if we need accurate info. And even then, it may take more work to get the unreferenced flag for unchanged datastores.

Reply:
Yes, I prefer that option too. We can update ISummaryStats to include this information and have every layer return it. Then we can get accurate count of number of DDSs as well. And yes, we can only get the unreferenced count for data stores that are summarized in this summary.
However, I was thinking we can get this information from the `collectGarbage` function. That computes the GC information so it does have access to `ids` of every data store and DDS that is referenced / unreferenced. We might need to do some extra work to correlate the `ids` to data store ids but it shouldn't be that hard.

Reply:
I don't think we will log number of dds this time. only dtastores